include/revng/Pipeline/LLVMContainer.h

```diff
@@ -142,49 +143,79 @@ class LLVMContainerBase
     auto BeforeEnumeration = this->enumerate();
     auto OtherEnumeration = Other.enumerate();
 
+    // We must ensure that merge(Module1, Module2).enumerate() ==
+    // merge(Module1.enumerate(), Module2.enumerate())
+    //
+    // So we enumerate now to have it later.
     BeforeEnumeration.merge(OtherEnumeration);
 
     ThisType &ToMerge = Other;
-    auto Composite = std::make_unique<llvm::Module>("llvm-link",
-                                                    Module->getContext());
-    std::set<std::string> Globals;
-
-    for (auto &Global : Module->globals()) {
-      if (Global.getLinkage() != llvm::GlobalValue::InternalLinkage)
-        continue;
-      Globals.insert(Global.getName().str());
-      Global.setLinkage(llvm::GlobalValue::ExternalLinkage);
+    std::set<std::string> Internals;
+    std::set<std::string> Externals;
+
+    // All symbols privated and not must be transformed into weak symbols, so
+    // that when multiple with the same name exists, one is dropped.
+    for (auto &Global : ToMerge.getModule().global_objects()) {
+
+      if (Global.getLinkage() == llvm::GlobalValue::InternalLinkage) {
+        Internals.insert(Global.getName().str());
+        Global.setLinkage(llvm::GlobalValue::LinkOnceODRLinkage);
```

Comment:
Please justify the choice of `LinkOnceODRLinkage`. I remember you had *reasons*.

Reply:
*cries in linker script*